sshbuf.c

```diff
@@ -109,6 +109,8 @@ sshbuf_set_parent(struct sshbuf *child, struct sshbuf *parent)
 	if ((r = sshbuf_check_sanity(child)) != 0 ||
 	    (r = sshbuf_check_sanity(parent)) != 0)
 		return r;
+	if (child->parent != NULL)
+		child->parent->refcount--;
```

Comment:
I don't think it makes sense for a child buffer to be "re-parented".

IMO we it would be better to explicitly disallow this, e.g.

```
        if (child->parent != NULL && child->parent != parent)
                return SSH_ERR_INTERNAL_ERROR;
```

Reply:
Yes, in that case it makes sense to treat it as internal error.